### ISSUE TITLE
Proposal ORKDrawingStep

### ISSUE THREAD

**Issue by oliverschaefer** [ResearchKit/ResearchKit#767]
## Motivation
Implement `ORKDrawingStep` which lets users draw any type of line-art with their finger/apple pencil
## Proposed Solution
Port [WWDC example] (https://developer.apple.com/library/prerelease/content/samplecode/SpeedSketch/Introduction/Intro.html) to ObjC and implement the corresponding `ORKDrawingStepViewController` and `ORKDrawingView`classes
## Impact on existing code
ORKSignatureStep could become a subclass of ORKDrawingStep. New implementation could fix #674 

@YuanZhu-apple @syoung-smallwisdom What do you think?

**Comment by syoung-smallwisdom** [comment]
I think it's a great idea.

**Comment by YuanZhu-apple** [comment]
@oliverschaefer Sounds good. My 2 cents, currently `ORKSignatureStep` carries no parameters, not sure it actually need to be the subclass of `ORKDrawingStep`.

**Comment by oliverschaefer** [comment]
@YuanZhu-apple Thanks for your input. Turns out, some aspects of SpeedSketch are not quite ready for real-word usage, see `CGDrawingEngine.swift` lines 137-144 for example. Could you help me do the port to ObjC and resolve these issues?

**Comment by YuanZhu-apple** [comment]
@oliverschaefer I think it basically equals to 

```
[array enumerateObjectsUsingBlock:^(id object, NSUInteger idx, BOOL *stop) {
    // do something with object
}];
```

**Comment by dillan** [comment]
@oliverschaefer I would love to see this happen as well. I think it would be particularly useful if the user was able to annotate an image. For example, allowing the user to annotate an anatomical model to indicate where they feel pain.

**Comment by oliverschaefer** [comment]
@dillan Yep, this was also the use case that I was thinking of. However, I'm currently working on printing support and the video instruction step and will only be able to work on the drawing step once my existing PRs are merged. This my take some time, since there currently isn't much activity going on. I guess the engineers in Cupertino are busy preparing for the macOS release.

**Comment by dillan** [comment]
It looks like the SpeedSketch sample code link above is no longer working. I believe it's the sample code for the [WWDC 2016 Session 220 - Leveraging Touch Input on iOS](https://developer.apple.com/videos/play/wwdc2016/220/). I'd be happy to help port it over to ObjC if anyone can link me to the code.

**Comment by oliverschaefer** [comment]
@dillan Thanks for your help - greatly appreciated. You're right seems like Apple removed the sample code for some reason. However, I have a local copy. Just drop me a line at oschaefer@me.com and i'll happily send that code over to you

**Comment by Pariecemckinney-apple** [comment]
This feature request has been recorded and will be tracked internally. Please feel free to open another issue if further discussion is needed.
